properties/test_index_manipulation.py

```diff
@@ -0,0 +1,156 @@
+import random
+
+import numpy as np
+import pytest
+
+from xarray import Dataset
+from xarray.indexes import PandasMultiIndex
+from xarray.testing import _assert_internal_invariants
+
+pytest.importorskip("hypothesis")
+
+import hypothesis.extra.numpy as npst
+import hypothesis.strategies as st
+from hypothesis import assume, note, settings
+from hypothesis.stateful import RuleBasedStateMachine, invariant, precondition, rule
+
+import xarray.testing.strategies as xrst
+
+
+def get_not_multiindex_dims(ds: Dataset) -> set:
+    dims = ds.dims
+    mindexes = [
+        name
+        for name, index in ds.xindexes.items()
+        if isinstance(index, PandasMultiIndex)
+    ]
+    return set(dims) - set(mindexes)
+
+
+def get_multiindex_dims(ds: Dataset) -> list:
+    mindexes = [
+        name
+        for name, index in ds.xindexes.items()
+        if isinstance(index, PandasMultiIndex)
+    ]
+    return mindexes
+
+
+def get_dimension_coordinates(ds: Dataset) -> set:
+    return set(ds.dims) & set(ds._variables)
+
+
+@st.composite
+def unique(draw, strategy):
+    # https://stackoverflow.com/questions/73737073/create-hypothesis-strategy-that-returns-unique-values
+    seen = draw(st.shared(st.builds(set), key="key-for-unique-elems"))
+    return draw(
+        strategy.filter(lambda x: x not in seen).map(lambda x: seen.add(x) or x)
+    )
+
+
+random.seed(123456)
+
+# Share to ensure we get unique names on each draw?
+UNIQUE_NAME = unique(strategy=xrst.names())
+DIM_NAME = xrst.dimension_names(name_strategy=UNIQUE_NAME, min_dims=1, max_dims=1)
```

Comment:
What do you mean by "share" here?

Reply:
I needed to draw a value from `names`   that is not present in the dataset for stack. 

Alternatively, I could use `assume(newname not in self.dataset)`